### PR TITLE
test: more CLI tests

### PR DESCRIPTION
**Pull request by artus9033**
This PR introduces the following:
- more integration tests to the `license-kit` CLI for the `report` command,
- unit tests for utilities in the shared package
- a CI workflow to run the unit tests
- fix for an edge case in `normalizeRepositoryUrl` (`git+ssh://git@github.com...` URLs would not be converted properly so far - they would have a `:` separator instead of `/`)
- clarifies a few details in the documentation of `license-kit`, corrects invalid default value & one typo in that documentation
- extracts helpers from `common.ts` to separate `yamlUtils.ts`
- adds badges for `test-integration-node.yml` & `test-unit-shared.yml` workflows to the root `README.md`

### CODE REVIEW

File: .github/workflows/test-unit-shared.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Integration tests - License Kit (Node)
```
Comment (mateusz1913) [inline .github/workflows/test-unit-shared.yml:1]: The name should be updated to sth like: "Unit tests - React Native Legal Shared package"

**Review by mateusz1913 — COMMENTED** [review]
Great job 🥳 left 2 comments to address before merging

File: .github/workflows/test-unit-shared.yml (hunk 1 above)
Comment (artus9033) [reply .github/workflows/test-unit-shared.yml]: Good find, thanks. Addressed in 8aeae76

File: README.md [hunk 2]
```diff
@@ -10,10 +10,12 @@
 
 ---
 
-![E2E tests - Android](https://github.com/callstackincubator/react-native-legal/actions/workflows/test-e2e-android.yaml/badge.svg)
-![E2E tests - iOS](https://github.com/callstackincubator/react-native-legal/actions/workflows/test-e2e-ios.yaml/badge.svg)
 ![Release](https://github.com/callstackincubator/react-native-legal/actions/workflows/release.yml/badge.svg)
 ![Deploy Docs](https://github.com/callstackincubator/react-native-legal/actions/workflows/deploy-docs.yml/badge.svg)
+![Integration tests - License Kit (Node)](https://github.com/callstackincubator/react-native-legal/actions/workflows/test-integration-node.yml/badge.svg)
+![Integration tests - React Native Legal Shared](https://github.com/callstackincubator/react-native-legal/actions/workflows/test-unit-shared.yml/badge.svg)
```
Comment (mateusz1913) [inline README.md:16]: Unit tests

File: .github/workflows/test-unit-shared.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+name: Integration tests - React Native Legal Shared package
```
Comment (mateusz1913) [inline .github/workflows/test-unit-shared.yml]: Unit tests

**Review by mateusz1913 — APPROVED** [review]
Great job! Two small things to change and let's ship it!

File: README.md (hunk 2 above)
Comment (artus9033) [reply README.md]: ... and this one I forgot. Addressed in cd87699

**Review by mateusz1913 — APPROVED** [review]
🚢
